src/storage/io/virtual_file_system.cppm

```diff
@@ -0,0 +1,96 @@
+// Copyright(C) 2024 InfiniFlow, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module;
+
+export module virtual_file_system;
+
+import stl;
+import file_system_type;
+import status;
+
+namespace infinity {
+
+export class VirtualFileSystem;
+
+export enum class FSType {
+    kLocal,
+    kMinio,
+    kAwsS3,
+    kAzureBlob, // Azure object store
+    kGCS,       // Google cloud storage
+    kOSS,       // Aliyun OSS
+    kCOS,       // Tencent object store
+    kOBS,       // Huawei object store
+    kHDFS,
+    kNFS,
+};
+
+export enum class FileAccessMode { kWrite, kRead, kInvalid };
+
+export class AbstractFileHandle {
+public:
```

Comment:
virtual destructor needed?
destructor calls Close if  Close is not called before.